docs/user/setup.rst

```diff
@@ -39,27 +39,45 @@ build time.
 Here are install instructions for a number of operating systems Scala
 Native has been used with:
 
-Ubuntu::
+**Ubuntu**
+::
+
+    $ sudo apt install clang libgc-dev libunwind-dev libre2-dev
 
-    $ sudo apt-get install clang libgc-dev libunwind-dev libre2-dev
+*Note:* libre2-dev is available since Xenial (16.04 LTS). Refer to `travis.yml <https://github.com/scala-native/scala-native/blob/master/.travis.yml>`_ to install from source.
 
-Note: libre2-dev is available since Xenial (16.04 LTS). Refer to `travis.yml <https://github.com/scala-native/scala-native/blob/master/.travis.yml>`_ to install from source.
+**Arch Linux**
+::
 
-macOS::
+    $ sudo pacman -S llvm gc re2
+
+**macOS**
+::
 
     $ brew install llvm bdw-gc re2
 
-FreeBSD::
+**FreeBSD**
+::
 
     $ pkg install llvm38 boehm-gc libunwind re2
 
-nix/nixOS::
+**Nix/NixOS**
+::
 
     $ wget https://raw.githubusercontent.com/scala-native/scala-native/master/bin/scala-native.nix
-
     $ nix-shell scala-native.nix -A clangEnv
 
 Continue to :ref:`sbt`.
 
+Troubleshooting
+---------------
+When compiling your Scala Native project, the linker ``ld`` may fail with the following message:
+
+::
+
+  relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a shared object; recompile with -fPIC
+
+It is likely that the ``LDFLAGS`` environment variable enables hardening. For example, this occurs when the ``hardening-wrapper`` package is installed on Arch Linux. It can be safely removed.
+
```

Comment:
I have an impression that this one deserves that this one belong on the page together with FAQ, given that we don't expect many people to hit this problem